modules/dnn/src/model.cpp

```diff
@@ -0,0 +1,265 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// Copyright (C) 2018-2019, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+
+#include "precomp.hpp"
+#include <algorithm>
+#include <iostream>
+#include <utility>
+#include <iterator>
+
+#include <opencv2/imgproc.hpp>
+
+namespace cv {
+namespace dnn {
+
+Model::Model(const std::string& model, const std::string& config, int width,
+      int height, Scalar mean, float scale, bool swapRB, bool crop) : Net(readNet(model, config)),
+      width_(width), height_(height), mean_(mean), scale_(scale), swapRB_(swapRB), crop_(crop) {}
+
+void Model::setInputWidth(int width) {
+    width_ = width;
+}
+
+void Model::setInputHeight(int height) {
+    height_ = height;
+}
+
+void Model::setInputMean(Scalar mean) {
+    mean_ = mean;
+}
+
+void Model::setInputScale(float scale) {
+    scale_ = scale;
+}
+
+void Model::setInputCrop(bool crop) {
+    crop_ = crop;
+}
+
+void Model::setInputSwapRB(bool swapRB) {
+    swapRB_ = swapRB;
+}
+
+std::pair<int, float> Model::classify(const Mat& frame) {
+    Mat blob = blobFromImage(frame, scale_, Size(width_, height_), mean_, swapRB_, crop_);
+    setInput(blob);
+    Mat out = forward();
+    std::vector<float> result = out.reshape(1, 1);
+    std::vector<float>::iterator iter = std::max_element(result.begin(), result.end());
+
+    std::pair<int, float> prediction(iter - result.begin(), *iter);
+    return prediction;
+}
+
+void Model::detect(Mat& frame, std::vector<int>& classIds,
+                   std::vector<float>& confidences, std::vector<Rect2d>& boxes,
+                   float confThreshold, float nmsThreshold)
+{
+    Mat blob = blobFromImage(frame, scale_, Size(width_, height_), mean_, swapRB_, crop_);
+    setInput(blob);
+    if (getLayer(0)->outputNameToIndex("im_info") != -1) {  // Faster-RCNN or R-FCN
+        resize(frame, frame, Size(width_, height_));
+        Mat imInfo = (Mat_<float>(1, 3) << frame.rows, frame.cols, 1.6f);
+        setInput(imInfo, "im_info");
+    }
+
+    std::vector<Mat> detections;
+    forward(detections);
```

Comment:
This method is https://docs.opencv.org/master/db/d30/classcv_1_1dnn_1_1Net.html#a8d248f50efec70dffa785de18a4b2170 but we want to use https://docs.opencv.org/master/db/d30/classcv_1_1dnn_1_1Net.html#adb34d7650e555264c7da3b47d967311b to get outputs from different layers (i.e. YOLOv3)